src/journal/JournalMetadata.cc

```diff
@@ -985,6 +991,8 @@ void JournalMetadata::committed(uint64_t commit_tid,
 
     ldout(m_cct, 20) << "updated commit position: " << commit_position << ", "
                      << "on_safe=" << m_commit_position_ctx << dendl;
+
+    disconnect_laggy_clients();
```

Comment:
I would only run this logic as part of the scheduled commit task.  The JournalTrimmer watches for header updates before it proceeds with the trim, so if it's part of the scheduled commit task, it could update the commit position, disconnect the laggy peers, and throw the update event for the Trimmer to detect the newly disconnected client.